### DISCUSSION TITLE
What's the differernces from next/core-web-vitals and eslint-config-next and eslint-plugin-next?

### DISCUSSION THREAD

**Discussion by zation** [vercel/next.js#58714] [Help]
### Summary

The document says every common eslint config is buildin. But there is no error for duplicated variables by default. The default eslint config is `eslint-config-next`, and the eslint document mentions `eslint-config-next` and `eslint-plugin-next`. This is really confused me. Can anyone please let me know what's the diferences from them? And what's the real working common eslint config for a nextjs project?

**Comment by 20manas** [comment]
First of all, when you put `{ "extends": "next" }` in eslint, eslint will be using config `eslint-config-next` (if it is installed).

When you do `{ "extends": "next/core-web-vitals" }`, eslint will use config  `eslint-config-next/core-web-vitals` which includes `eslint-config-next`

So, the `eslint-config-next/core-web-vitals` config includes:
- `eslint-config-next`
- core web vitals plugin which includes core web vitals rules

`eslint-config-next` includes:
- `eslint-plugin-next`
- `eslint-plugin-react`
- `eslint-plugin-react-hooks`

`eslint-plugin-next` contains next js specific rules

What this means is that `eslint-config-next/core-web-vitals` is the strictest next config and includes everything.

However, it does not include basic eslint configs like `eslint:recommended`. So things like duplicate variables don't throw an error. So configs outside the realm of next and react have to be included separately.

The eslint config extends that I personally use:

```js
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/strict-type-checked',
    'plugin:@typescript-eslint/stylistic-type-checked',
    'next/core-web-vitals',
    'prettier',
  ],
```

- [typescript-eslint config](https://typescript-eslint.io/): typescript linting. Can be overkill. 
- [prettier config](https://github.com/prettier/eslint-config-prettier): if you are using prettier, this config can be used so that eslint does not conflict with prettier

**Reply by gnowland** [reply]
This should be in the official docs @vercel

**Comment by ghost** [comment]
Would be great to get an official response from @vercel. At least currently it seems like installing the required packages for linting typescript is not possible.

Trying to add the recommended extends from above results in the following error:
```
Oops! Something went wrong! :(

ESLint: 8.57.0

ESLint couldn't find the plugin "@typescript-eslint/eslint-plugin".

(The package "@typescript-eslint/eslint-plugin" was not found when loaded as a Node module from the directory "/Projects/next".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install @typescript-eslint/eslint-plugin@latest --save-dev

The plugin "@typescript-eslint/eslint-plugin" was referenced from the config file in ".eslintrc.json".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.

husky - pre-commit script failed (code 1)
```

Running the recommended command results in a conflict.
```
npm install @typescript-eslint/eslint-plugin --save-dev
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: next@0.1.0
npm ERR! Found: @typescript-eslint/parser@7.2.0
npm ERR! node_modules/@typescript-eslint/parser
npm ERR!   @typescript-eslint/parser@"^5.4.2 || ^6.0.0 || 7.0.0 - 7.2.0" from eslint-config-next@14.2.3
npm ERR!   node_modules/eslint-config-next
npm ERR!     dev eslint-config-next@"14.2.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! dev @typescript-eslint/eslint-plugin@"*" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @typescript-eslint/parser@8.3.0
npm ERR! node_modules/@typescript-eslint/parser
npm ERR!   peer @typescript-eslint/parser@"^8.0.0 || ^8.0.0-alpha.0" from @typescript-eslint/eslint-plugin@8.3.0
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /.npm/_logs/2024-08-29T05_04_06_990Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /.npm/_logs/2024-08-29T05_04_06_990Z-debug-0.log
